packages/graph/src/graph2/layouts/dagre.ts

```diff
@@ -14,21 +14,38 @@ export class Dagre extends Layout {
         super.start();
         const size = this._graph.size();
         const data = this._graph.graphData();
-
+        const nodeIndexMap = {};
+        const nodes = data.vertices().map((v,i) => {
+            const bbox = v.element.node().getBBox();
+            nodeIndexMap[v.id] = i;
+            return {
+                vertex_id: v.id,
+                width: bbox.width,
+                height: bbox.height,
+                ...v.props,
+                id: i
+            };
+        });
+        const links = data.edges().map((e,i) => {
+            const source = {...e.props.source};
+            const target = {...e.props.target};
```

Comment:
Do source and target need their props?  Actually I suspect 
```
const source = nodeIndexMap[e.props.source..id]
const target = nodeIndexMap[e.props.target..id]
```
Would be better (with your undefined check)?;